qa/L0_python_api/testing_utils.py

```diff
@@ -81,15 +83,63 @@ def send_and_test_inference_identity(frontend_client, url: str) -> bool:
     input_data = np.array(["testing"], dtype=object)
 
     # Create input and output objects
-    inputs = [frontend_client.InferInput("INPUT0", input_data.shape, "BYTES")]
-    outputs = [frontend_client.InferRequestedOutput("OUTPUT0")]
+    inputs = [frontend_client.InferInput("text_input", input_data.shape, "BYTES")]
+    outputs = [frontend_client.InferRequestedOutput("text_output")]
     # Set the data for the input tensor
     inputs[0].set_data_from_numpy(input_data)
 
     # Perform inference request
     results = client.infer(model_name=model_name, inputs=inputs, outputs=outputs)
 
-    output_data = results.as_numpy("OUTPUT0")  # Gather output data
+    output_data = results.as_numpy("text_output")  # Gather output data
 
     teardown_client(client)
     return input_data[0] == output_data[0].decode()
+
+
+# Sends a streaming inference request to test_model_repository/identity model
+# and verifies input == output
+def send_and_test_stream_inference(frontend_client, url: str) -> bool:
+    model_name = "identity"
+
+    # Setting up the gRPC client stream
+    results = queue.Queue()
+    callback = lambda error, result: results.put(error or result)
+    client = frontend_client.InferenceServerClient(url=url)
+
+    client.start_stream(callback=callback)
+
+    # Preparing Input Data
+    text_input = "testing"
+    input_tensor = frontend_client.InferInput(
+        name="text_input", shape=[1], datatype="BYTES"
+    )
+    input_tensor.set_data_from_numpy(np.array([text_input.encode()], dtype=np.object_))
+
+    # Sending Streaming Inference Request
+    client.async_stream_infer(
+        model_name=model_name, inputs=[input_tensor], enable_empty_final_response=True
+    )
+
+    # Looping through until exception thrown or request completed
+    completed_requests, num_requests = 0, 1
+    text_output, is_final = None, None
+    while completed_requests != num_requests:
+        result = results.get()
+        if isinstance(result, InferenceServerException):
+            if result.status() == "StatusCode.CANCELLED":
+                completed_requests += 1
+            raise result
+
+        # Processing Response
+        text_output = result.as_numpy("text_output")[0].decode()
+        is_final = result.get_response().parameters.get("triton_final_response", False)
```

Comment:
I believe this should be something like 
```python
result.get_response().parameters.get("triton_final_response", False).bool_param
```

Without the `bool_param` part, you might get some truthy value (non-empty dict) that looks like `{"bool_param": False}` or something, and get unexpected behavior.

Reply:
See https://github.com/triton-inference-server/triton_cli/blob/36a425e74c91402e7a6c9ac818aa49f3a5c41674/src/triton_cli/client/client.py#L288-L297